secio/src/crypto/ring_impl.rs

```diff
@@ -130,6 +142,15 @@ impl StreamCipher for RingAeadCipher {
     fn decrypt(&mut self, input: &[u8]) -> Result<Vec<u8>, SecioError> {
         self.decrypt(input)
     }
+
+    #[inline]
+    fn is_in_place(&self) -> bool {
+        true
+    }
+
+    fn decrypt_in_place(&mut self, input: &mut bytes::BytesMut) -> Result<(), SecioError> {
```

Comment:
If no new fn is added, OpenSSL will copy one more time, because OpenSSL bind does not have an in-place interface

Reply:
At the same time, because `&mut [u8]` does not have shortened semantics, the delete tag operation on `in-place` cannot be performed on this type. Forcibly shortening its length with a pointer will cause a dangling pointer, so the type must be marked with shortening semantics type

Reply:
how about change to BytesMut?
```
fn decrypt(&mut self, input: &mut BytesMut) -> Result<(), SecioError>
```